tests/RAGchain/DB/test_base_db.py

```diff
@@ -107,3 +122,10 @@ def search_test_base(db: BaseDB):
     assert len(test_result_11) == 2
     assert 'test_id_2' in [passage.id for passage in test_result_11]
     assert 'test_id_4' in [passage.id for passage in test_result_11]
+
+
+def duplicate_id_test_base(db: BaseDB, error_type: Type[Exception]):
+    with pytest.raises(error_type):
+        db.save(DUPLICATE_PASSAGE)
+    db.save(DUPLICATE_PASSAGE, upsert=True)
+    assert db.fetch([DUPLICATE_PASSAGE[0].id]) == DUPLICATE_PASSAGE
```

Comment:
add new line (press Enter)

Reply:
Plus, you need to turn on 'reformat code' and 'optimize import' at pycharm commit section. You can find right bottom setting button next to commit&push button in pycharm.

Reply:
thx for your kind Additional Comments 👍